modules/cudaoptflow/include/opencv2/cudaoptflow.hpp

```diff
@@ -342,6 +383,69 @@ class CV_EXPORTS_W OpticalFlowDual_TVL1 : public DenseOpticalFlow
             bool useInitialFlow = false);
 };
 
+//
+// NvidiaOpticalFlow
+//
+
+/** @brief Class for computing the optical flow vectors between two images using NVIDIA Optical Flow hardware and Optical Flow SDK 1.0.
+@note
+- A sample application demonstrating the use of NVIDIA Optical Flow can be found at
+opencv_source_code/samples/gpu/nvidia_optical_flow.cpp
+- An example application comparing accuracy and performance of NVIDIA Optical Flow with other optical flow algorithms in OpenCV can be found at
+opencv_source_code/samples/gpu/optical_flow.cpp
+*/
+
+/**
+* Supported optical flow/stereo disparity performance levels.
+*/
+typedef enum _NVIDIA_OF_PERF_LEVEL
```

Comment:
NVIDIA_OF_PERF_LEVEL enum is not needed in NvidiaOpticalFlow_1_0 right? since it is casted to NV_OF_PERF_LEVEL

Reply:
Done.

Reply:
If i move it inside class then everywhere i need to refer it like this cuda::NvidiaOpticalFlow_1_0::NVIDIA_OF_PERF_LEVEL otherwise it gives build errors.